7 spring cloud microservices

RinatkaDyrov opened the pull request.

RinatkaDyrov committed:
initial commit

RinatkaDyrov committed:
7-spring-cloud-microservices
eureka theme completed

RinatkaDyrov committed:
7-spring-cloud-microservices
eureka theme completed

Commit:
7-spring-cloud-microservices
start

RinatkaDyrov committed:
7-spring-cloud-microservices
final task in process

RinatkaDyrov committed:
7-spring-cloud-microservices
shopping-store done

Commit:
7-spring-cloud-microservices
shopping-store done
tests passed

RinatkaDyrov committed:
7-spring-cloud-microservices
shopping-store done
shopping-cart in process

RinatkaDyrov committed:
7-spring-cloud-microservices
shopping-store done
shopping-cart in process_2

Commit:
7-spring-cloud-microservices
shopping-store done
shopping-cart done
warehouse done
postman tests were passed

Commit:
7-spring-cloud-microservices
shopping-store done
shopping-cart done
warehouse done
postman tests were passed

VladimirTitov4 commented on infra/config-server/src/main/resources/application.yml:
```diff
@@ -9,5 +9,10 @@ spring:
 server:
-  port: 8888
\ No newline at end of file
+  port: 0
```
Порт 0 означает случайный порт, что может усложнить отладку. Рекомендуется использовать фиксированные порты или порты из диапазона. 

Вы точно хотите здесь оставить случаный ?  (возможно решение верное, просто прокомментируйте)

RinatkaDyrov replied:
ну там нет такого условия по заданию, но нулевой порт приводится как удобный способ генерации случайного порта: "Чтобы микросервисы могли работать вместе, важно, чтобы все они запускались на разных портах. Для этого удобно настроить случайный выбор порта для каждого из микросервисов — достаточно указать параметр server.port в конфигурации микросервисов равным 0. В результате порт будет генерироваться случайным образом, а взаимодействие микросервисов будет обеспечено через Eureka."

VladimirTitov4 commented on infra/config-server/src/main/resources/config/commerce/shopping-cart/application.yml:
```diff
@@ -0,0 +1,23 @@
+    password: cart
+
+server:
+  port: 0
```
И здесь

RinatkaDyrov replied:
аналогично)

VladimirTitov4 commented on commerce/shopping-cart/src/main/java/ru/yandex/practicum/ShoppingCartApp.java:
```diff
@@ -0,0 +1,14 @@
+import ru.yandex.practicum.feignClient.WarehouseClient;
+
+@SpringBootApplication
+@EnableFeignClients(clients = WarehouseClient.class)
```
Желательно записывать как то так

@EnableFeignClients(basePackages = "ru.yandex.practicum.feignClient")

RinatkaDyrov replied:
Океееей, понял, принял, поправил)

RinatkaDyrov committed:
shopping-store done
shopping-cart done
warehouse done
postman tests were passed
EnableFeignClients annotation fixed
